src/transactions/test/ClaimableBalanceTests.cpp

```diff
@@ -0,0 +1,1040 @@
+
+                        predicateTest(connectPredicate(false, relAfterFalse,
+                                                       connectedRight));
+                    }
+                    SECTION("complex 3")
+                    {
+                        // full tree with all ANDs. All predicates are satisfied
+                        auto tree1 =
+                            connectPredicate(true, absAfterTrue, absBeforeTrue);
+                        auto tree2 =
+                            connectPredicate(true, relAfterTrue, relBeforeTrue);
+                        auto tree3 =
+                            connectPredicate(true, absAfterTrue, relBeforeTrue);
+                        auto tree4 =
+                            connectPredicate(true, relAfterTrue, absBeforeTrue);
+
+                        auto c1 = connectPredicate(true, tree1, tree2);
+                        auto c2 = connectPredicate(true, tree3, tree4);
+                        predicateTest(connectPredicate(true, c1, c2));
+                    }
+                    SECTION("complex 4")
+                    {
+                        // full tree with all ORs. One predicate is satisfied
+                        auto tree1 = connectPredicate(false, absAfterFalse,
+                                                      absBeforeFalse);
+                        auto tree2 = connectPredicate(false, relAfterFalse,
+                                                      relBeforeFalse);
+                        // absAfterTrue is used here
+                        auto tree3 = connectPredicate(false, absAfterTrue,
+                                                      relBeforeFalse);
+                        auto tree4 = connectPredicate(false, relAfterFalse,
+                                                      absBeforeFalse);
+
+                        auto c1 = connectPredicate(false, tree1, tree2);
+                        auto c2 = connectPredicate(false, tree3, tree4);
+                        predicateTest(connectPredicate(false, c1, c2));
+                    }
+                }
+
+                SECTION("balance does not exist")
+                {
+                    acc1.createClaimableBalance(asset, amount, validClaimants);
+                    REQUIRE_THROWS_AS(
+                        acc1.claimClaimableBalance(ClaimableBalanceID{}),
+                        ex_CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);
+                }
+
+                SECTION("no destination match")
+                {
+                    auto simpleBalanceID = acc1.createClaimableBalance(
+                        asset, amount, validClaimants);
+                    // issuer is not one of the claimants
+                    REQUIRE_THROWS_AS(
+                        issuer.claimClaimableBalance(simpleBalanceID),
+                        ex_CLAIM_CLAIMABLE_BALANCE_CANNOT_CLAIM);
+                }
+            }
+
+            SECTION("multiple create and claims")
```

Comment:
Not sure I understand the purpose of this test? Unless I'm mistaken, it effectively does "create -> claim -> fund -> create -> claim" with each step in its own transaction. Was the intention to do that as a single transaction?

Reply:
The test here was to make sure that the claim doesn't have any unintended side effects that could get in the way of another create/claim later on.